util/src/main/java/tech/pegasys/artemis/util/async/SafeFuture.java

```diff
@@ -67,15 +67,24 @@ public static void reportExceptions(final CompletionStage<?> future) {
   public static <U> SafeFuture<U> of(final Supplier<CompletionStage<U>> futureSupplier) {
```

Comment:
nit: I think we can get rid of this one and just have the `ExceptionThrowingSupplier` versions.